tests/lib/rules/no-restricted-exports.js

```diff
@@ -533,6 +537,44 @@ ruleTester.run("no-restricted-exports", rule, {
             ]
         },
 
+        // restrictedNamedExportsPattern
+        {
+            code: "var getSomething; export { getSomething };",
+            options: [{ restrictedNamedExportsPattern: "get*" }],
+            errors: [
+                { messageId: "restrictedNamed", data: { name: "getSomething" }, type: "Identifier" }
+            ]
+        },
+        {
+            code: "var getSomethingFromUser; export { getSomethingFromUser };",
+            options: [{ restrictedNamedExportsPattern: "User$" }],
+            errors: [
+                { messageId: "restrictedNamed", data: { name: "getSomethingFromUser" }, type: "Identifier" }
+            ]
+        },
+        {
+            code: "var foo, ab, xy; export { foo, ab, xy };",
+            options: [{ restrictedNamedExportsPattern: "(b|y)$" }],
+            errors: [
+                { messageId: "restrictedNamed", data: { name: "ab" }, type: "Identifier" },
+                { messageId: "restrictedNamed", data: { name: "xy" }, type: "Identifier" }
+            ]
+        },
+        {
+            code: "var foo; export { foo as ab };",
+            options: [{ restrictedNamedExportsPattern: "(b|y)$" }],
+            errors: [
+                { messageId: "restrictedNamed", data: { name: "ab" }, type: "Identifier" }
+            ]
+        },
+        {
+            code: "var privateUserEmail; export { privateUserEmail };",
+            options: [{ restrictedNamedExportsPattern: "^privateUser" }],
+            errors: [
+                { messageId: "restrictedNamed", data: { name: "privateUserEmail" }, type: "Identifier" }
+            ]
+        },
+
```

Comment:
can you also add an example with negation `!` pattern, like one given in the issue?